lib/charms/layer/zookeeper.py

```diff
@@ -97,10 +111,29 @@ def start(self):
         zookeeper_home = self.dist_config.path('zookeeper')
         self.stop()
         utils.run_as('zookeeper', '{}/bin/zkServer.sh'.format(zookeeper_home), 'start')
+        self.start_rest()
 
     def stop(self):
         zookeeper_home = self.dist_config.path('zookeeper')
         utils.run_as('zookeeper', '{}/bin/zkServer.sh'.format(zookeeper_home), 'stop')
+        self.stop_rest()
+
+    def start_rest(self):
+        self.stop_rest()
+        zookeeper_rest = self.dist_config.path('zookeeper') / 'src/contrib/rest'
+        zkrest_logs = self.dist_config.path('zookeeper_log_dir') / 'rest.out'
+        zkrest_buildxml = zookeeper_rest / 'build.xml'
+        self.run_bg('zookeeper', zkrest_logs, 'nohup', 'ant', 'run', '-f', zkrest_buildxml)
+
+    def stop_rest(self):
+        print("*** Stopping rest ***")
+        try:
+            # Give some time to the java process to spawn
+            # in the case of a start followed by an immediate stop
+            sleep(30)
+            utils.run_as('root', 'pkill', '-f', 'RestMain')
+        except:
+            pass
```

Comment:
This makes me a sad panda.  :(  You should never unconditionally mask any possible error like this.  If we hit something unexpected, like `pkill` not being available by default on Xenial, we could end up spending a long time tracking this down.

At best, this should be restricted to only catching `CalledProcessError` and / or `OSError` and should at a minimum log it.  But in this case, I think the better approach is to guard this with a `pgrep` as mentioned above.